### PR TITLE
Skip an object if subsequent api calls fail for that object

### PR DESCRIPTION
**Pull request by fahia**
### What it does and why:
- When building an object requires multiple requests, fail and move onto the next object if any of the calls happen to fail. Log this to stderr to let the user know.

### Checklist:

- [x] Testing & QA: Passes tests and linting.
- [x] :tophat:: I have manually tested these changes.

---

### Related issues:
Closes: https://github.com/Shopify/shopify_transporter/issues/68, https://github.com/Shopify/shopify_transporter/issues/67

### CODE REVIEW

File: lib/shopify_transporter/exporters/magento/soap.rb [hunk 1]
```diff
@@ -33,10 +33,14 @@ def call_in_batches(method:, params: {}, batch_index_column:)
...
-                method,
-                params.merge(batching_filter(current_id, end_of_range, batch_index_column)),
-              )
+              begin
+                enumerator << call(
+                  method,
+                  params.merge(batching_filter(current_id, end_of_range, batch_index_column)),
+                )
+              rescue Savon::Error
+                $stderr.puts "Skipping batch: #{current_id}..#{end_of_range} after #{MAX_RETRIES} retries because of an error."
```
Comment (surajreddy) [inline lib/shopify_transporter/exporters/magento/soap.rb]: Do you mind adding the error details here as well?
```
 $stderr.puts 'The exact error was:'
... e.class + e.message
```

File: lib/shopify_transporter/exporters/magento/customer_exporter.rb [hunk 2]
```diff
@@ -38,6 +43,20 @@ def base_customers
         def customer_address_list(customer_id)
           @client.call(:customer_address_list, customer_id: customer_id).body
         end
+
+        def print_customer_details_error(customer, e)
```
Comment (surajreddy) [inline lib/shopify_transporter/exporters/magento/customer_exporter.rb]: Do you anticipate the error handling to change for these?

The output looks very similar for all these objects, so I'm wondering if we should extract the method to a helper class. I don't feel too strongly about this, so as is is fine as well.

Comment (fahia) [reply lib/shopify_transporter/exporters/magento/customer_exporter.rb]: How's moving it to the base exporter class?

File: lib/shopify_transporter/exporters/magento/soap.rb (hunk 1 above)
Comment (fahia) [reply lib/shopify_transporter/exporters/magento/soap.rb]: Sure!

File: lib/shopify_transporter/exporters/magento/customer_exporter.rb (hunk 2 above)
Comment (surajreddy) [reply lib/shopify_transporter/exporters/magento/customer_exporter.rb]: I'm okay with that - would we lose functionality that way?

Comment (fahia) [reply lib/shopify_transporter/exporters/magento/customer_exporter.rb]: Yeah, it'd end up being less friendly. I went ahead and tried that out but realized it actually worsens the experience. I prefer this slightly less DRY approach that can easily be changed later /shrug.

Comment (JordanVerasamy) [reply lib/shopify_transporter/exporters/magento/customer_exporter.rb]: agree w fahia

**Comment by fahia** [comment]
Rubocop fails because of the begin/rescue issue. Tests are all green

File: spec/shopify_transporter/exporters/magento/soap_spec.rb [hunk 3]
```diff
@@ -186,6 +186,36 @@ def expected_batching_filter(batch_key, batch_range_string)
...
+                'customer_id',
+                '3,4,5'
+              )) do
+              raise Savon::Error, 'Soap call failed.'
+            end
+
+
+            expect { |b| soap_client.call_in_batches(method: :test_call, batch_index_column: 'customer_id').each(&b) }
+              .to yield_successive_args(['batch-0,1,2'], ['batch-6,7'])
+          end
```
Comment (surajreddy) [inline spec/shopify_transporter/exporters/magento/soap_spec.rb]: We should add a check here for the message output to $stderr as well, similar to the other tests.

Comment (fahia) [reply spec/shopify_transporter/exporters/magento/soap_spec.rb]: good call, forgot about this

**Pull request merged**